SwiftyBluetooth/Source/CentralProxy.swift

```diff
@@ -144,7 +144,7 @@ extension CentralProxy {
         let weakRequest = timer.userInfo as! Weak<PeripheralScanRequest>
         
         if weakRequest.value != nil {
-            self.stopScan()
```

Comment:
For example, if the user is only looking for one peripheral, once it has been found `.stopScan()` should be called, rather than letting the scan continue until timer completion. I don't see how this can be easily done with the library's current functionality as it isn't possible to distinguish between the `.stopScan()` called by the scanner timing out, vs. when it is called by the user. It also just seems odd to treat a timeout as an error in some cases and not in others?